### PR TITLE
Akka.Cluster.Tools: deprecate ClustersSingletonManagerSettings.ConsiderAppVersion

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Cleanup from PR https://github.com/akkadotnet/akka.net/pull/7298

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #7197
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs [hunk 1]
```diff
@@ -619,6 +575,17 @@ public ClusterSingletonManager(Props singletonProps, object terminationMessage,
...
                 _leaseRetryInterval = settings.LeaseSettings.LeaseRetryInterval;
             }
+            
+            // Added in v1.5.27 to signal to users who were considering AppVersion
+            // in their singleton placement decisions that we don't do that any more
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (settings.ConsiderAppVersion)
+#pragma warning restore CS0618 // Type or member is obsolete
+            {
+                Log.Warning("As of Akka.NET v1.5.27, The 'ConsiderAppVersion' setting is no longer supported and will " +
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs:585]: Add a warning for users who had `ConsiderAppVersion` enabled - it was disabled by default, so it shouldn't affect too many people.

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManagerSettings.cs [hunk 2]
```diff
@@ -188,7 +189,9 @@ private static string RoleOption(string role)
             RemovalMargin = removalMargin;
             HandOverRetryInterval = handOverRetryInterval;
             LeaseSettings = leaseSettings;
+#pragma warning disable CS0618 // Type or member is obsolete
             ConsiderAppVersion = considerAppVersion;
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManagerSettings.cs:193]: Mark the setting as obsolete - it's already not getting used.

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs:1]: The rest of the code changes in this file are fixing a few linger nullability issues and cleaning up `TBD` XML-DOC comments.

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
A remember-entities spec failed - I'll have to dig into that one later.

**Pull request merged**